### DISCUSSION TITLE
Custom JS logic debugging

### DISCUSSION THREAD

**Discussion by AdrianDengusiak** [SAP/open-ux-odata#481] [Q&A] [answered]
Hi, I started to use this middleware for freestyle SAPUI5 apps and it works perfecly, except one thing (I'm probably not sure how to proceed).

I have JS module for my entity set, let's say **UserSet.js**, located in webapp/localService/mockdata. The problem is that it's not being loaded in the **Sources** tab in Chrome DevTools. I also can't find this JS module anywhere, so it's not possible to setup a breakpoint and debug API functions/params. I know module works, because it modifies dataset in my app. I tried setting up **debug** flag for middleware, but it didn't help.

I would be grateful if you can help or at least explain how it works.

Adrian

**Comment by nlunets** [comment, marked as answer]
Hi @AdrianDengusiak,

There is no reason for it to be loaded in the chrome dev tools as the mockserver runs as a node process, not in your browser.

What you need to do is from your IDE, start the application process in debug mode and put breakpoint there. There should be plenty of guide on how this can be achieved depending on your IDE of choice.

That should allow you to debug directly !

**Reply by AdrianDengusiak** [reply]
You're totally right. I launched my script in debug mode in VSCode and caught required breakpoint.
Thanks a lot.
